internal/services/extendedlocation/extended_location_custom_location.go

```diff
@@ -0,0 +1,318 @@
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			if model := resp.Model; model != nil {
+				props := model.Properties
+
+				state := CustomLocationResourceModel{
+					Name:              id.CustomLocationName,
+					ResourceGroupName: id.ResourceGroupName,
+					Location:          model.Location,
+				}
+
+				if props.ClusterExtensionIds != nil {
+					state.ClusterExtensionIds = pointer.From(props.ClusterExtensionIds)
+				}
+				if props.DisplayName != nil {
+					state.DisplayName = pointer.From(props.DisplayName)
+				}
+				if props.HostResourceId != nil {
+					state.HostResourceId = pointer.From(props.HostResourceId)
+				}
+				if props.HostType != nil {
+					state.HostType = string(pointer.From(props.HostType))
+				}
+				if props.Namespace != nil {
+					state.Namespace = pointer.From(props.Namespace)
+				}
+
+				if props.Authentication != nil {
+					state.Authentication = []AuthModel{
+						{
+							Type:  pointer.From(props.Authentication.Type),
+							Value: pointer.From(props.Authentication.Value),
+						},
+					}
+				}
+
+				return metadata.Encode(&state)
+			}
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("deleting %s", *id)
+
+			if resp, err := client.Delete(ctx, *id); err != nil {
+				if !response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("deleting %s: %+v", *id, err)
+				}
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r CustomLocationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ExtendedLocation.CustomLocationsClient
+			id, err := customlocations.ParseCustomLocationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state CustomLocationResourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			customLocationProps := customlocations.CustomLocationProperties{}
+			d := metadata.ResourceData
+
+			if d.HasChanges("authentication") {
+				if state.Authentication != nil && len(state.Authentication) > 0 {
```

Comment:
No need for a nil check here, the length check will suffice

```suggestion
				if len(state.Authentication) > 0 {
```